src/main/java/com/eleks/academy/whoami/service/impl/GameServiceImpl.java

```diff
@@ -118,10 +134,49 @@ public void submitGuess(String gameId, String player, Message guess) {
         }
     }
 
+
+    @Override
+    public void answerGuessingQuestion(String gameId, String playerId, QuestionAnswer answerQuess) {
+        PersistentGame game = checkGameExistence(gameId);
+        if (game.getStatus().equals(GameStatus.GAME_IN_PROGRESS)) {
+            game.answerGuessingQuestion(playerId, answerQuess);
+        }
+    }
+
     private PersistentGame checkGameExistence(String gameId) {
         if (gameRepository.findById(gameId).isPresent()) {
             return gameRepository.findById(gameId).get();
         }
         throw new GameNotFoundException(String.format(ROOM_NOT_FOUND_BY_ID, gameId));
     }
+
+    @Override
+    public String gameHistory(String gameId) {
+        PersistentGame game = checkGameExistence(gameId);
+        return (new HistoryDetails(game.getHistory())).toString();
```

Comment:
Same as above, why wrap it in additional class and then just call toString()? Just 
`return game.getHistory() `
will be enough and FE will get- all data that they need and will format it as they want.
You don't need to format strings for representation - it is job for Front-end. Back-end goal is to return correct data